client/src/graphs/svg/renderers/ProvenanceRenderer.ts

```diff
@@ -0,0 +1,128 @@
+import * as d3 from 'd3';
+
+import { SVGRenderer } from 'svg-flowgraph';
+
+import { SVGRendererOptionsInterface } from '@/types/typesGraphs';
+
+import { flatten } from '@/graphs/svg/util';
+import { Colors } from '@/graphs/svg/encodings';
+import SVGUtil from '@/utils/SVGUtil';
+
+const pathFn = SVGUtil.pathFn().curve(d3.curveBasis);
+
+const DEFAULT_STYLE = {
+  node: {
+    fill: Colors.NODES.DEFAULT,
+    stroke: Colors.STROKE,
+    strokeWidth: 2,
+    borderRadius: 5,
+    controlSize: 15,
+    controlColor: Colors.CONTAINER_CONTROL,
+  },
+  edge: {
+    fill: 'none',
+    stroke: Colors.EDGES.DEFAULT,
+    strokeWidth: 5,
+    controlRadius: 6,
+    controlStrokeWidth: 2,
+    controlStrokeColor: Colors.NODES.DEFAULT,
+  },
+};
+
+export default class ProvenanceRenderer extends SVGRenderer {
+  constructor (options:SVGRendererOptionsInterface) {
+    super(options);
+  }
+
+  static calcNodeColor (d: d3.Selection<any, any, any, any>): string {
+    return (d as any).nodes ? Colors.NODES.CONTAINER : DEFAULT_STYLE.node.fill;
+  }
+
+  buildDefs (): void {
+    const svg = d3.select((this as any).svgEl);
+    const edges = flatten((this as any).layout).edges;
+
+    // Clean up
+    svg.select('defs').selectAll('.edge-marker-end').remove();
+
+    // Arrowheads for edges
+    svg.select('defs')
+      .selectAll('.edge-marker-end')
+      .data(edges)
+      .enter()
+      .append('marker')
+      .classed('edge-marker-end', true)
+      .attr('id', d => {
+        const source = (d as any).source.replace(/\s/g, '');
+        const target = (d as any).target.replace(/\s/g, '');
+        return `arrowhead-${source}-${target}`;
+      })
+      .attr('viewBox', SVGUtil.MARKER_VIEWBOX)
+      .attr('refX', 2)
+      .attr('refY', 0)
+      .attr('orient', 'auto')
+      .attr('markerWidth', 15)
+      .attr('markerHeight', 15)
+      .attr('markerUnits', 'userSpaceOnUse')
+      .attr('xoverflow', 'visible')
+      .append('svg:path')
+      .attr('d', SVGUtil.ARROW)
+      .style('fill', Colors.EDGES.DEFAULT)
+      .style('stroke', 'none');
+  }
+
+  renderNode (nodeSelection: d3.Selection<any, any, any, any>): void {
+    nodeSelection.each(function () {
+      const selection = d3.select(this);
+      const datum = selection.datum();
+
+      // Operations are displayed as rectangles while models, parameters, and variables are displayed as ellipses
+      if (!(datum as any).data.role?.includes('Operation')) {
```

Comment:
I think you are actually doing the opposite of what you say in the comment, I think you need to remove the logical not (!) and that should do it.

Reply:
Thanks for catching this!